### PR TITLE
Adoption and stewardship report routes

### PR DESCRIPTION
**Pull request by chromium-52**
[Clickup ticket](https://app.clickup.com/t/1q8byxp)

Created the report router and added routes that return adoption and stewardship reports.

Updating documentation in https://github.com/Code-4-Community/c4c-dev-docs/pull/90.

(Potentially also considering implementing the CSV adoption/stewardship report routes.)

### CODE REVIEW

File: service/src/main/java/com/codeforcommunity/processor/ProtectedReportProcessorImpl.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                .leftJoin(USERS)
+                .on(ADOPTED_SITES.USER_ID.eq(USERS.ID))
+                .leftJoin(STEWARDSHIP)
+                .on(ADOPTED_SITES.SITE_ID.eq(STEWARDSHIP.SITE_ID))
+                .leftJoin(NEIGHBORHOODS)
+                .on(SITES.NEIGHBORHOOD_ID.eq(NEIGHBORHOODS.ID))
+                .groupBy(SITES.ID, SITES.ADDRESS, USERS.FIRST_NAME, USERS.LAST_NAME, USERS.EMAIL, ADOPTED_SITES.DATE_ADOPTED,
+                        NEIGHBORHOODS.NEIGHBORHOOD_NAME)
+                .orderBy(USERS.FIRST_NAME, USERS.LAST_NAME, ADOPTED_SITES.DATE_ADOPTED, SITES.ID)
+                .fetchInto(AdoptedSite.class);
```
Comment (willmt80) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedReportProcessorImpl.java:60]: 💯

**Review by willmt80 — APPROVED** [review]
🔥 This looks good to me!

File: service/src/main/java/com/codeforcommunity/processor/ProtectedReportProcessorImpl.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    /**
+     * Throws an exception if the user is not an admin or super admin.
+     *
+     * @param level the privilege level of the user calling the route
+     */
+    void isAdminCheck(PrivilegeLevel level) {
+        if (!(level.equals(PrivilegeLevel.ADMIN) || level.equals(PrivilegeLevel.SUPER_ADMIN))) {
+            throw new AuthException("User does not have the required privilege level.");
```
Comment (SofieDunt) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedReportProcessorImpl.java:38]: ⛏️ Might be more helpful for the user if the message specified they had to be an admin.

**Review by SofieDunt — APPROVED** [review]
Looks good to me! 💯

**Review by vrushalitarte — APPROVED** [review]

**Pull request merged**